### ISSUE TITLE
Is there any way to use express router with RegisterRoutes?

### ISSUE THREAD

**Issue by zenmaster15** [lukeautry/tsoa#677]
So i'm trying to implement versioning in my app using endpoints with /v1 and /v2. But i'm kind of confused on how to use express.router() on RegisterRoutes. Is it possible? or how should i move about on implementing multiple api versions with tsoa?

## Sorting

- **I'm submitting a ...**

  - [ ] bug report
  - [ ] feature request
  - [x] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

**Comment by WoH** [comment]
We currently do not expose the express router and do not intend to.
Given the very limited information provided for a very nuanced topic, I think you want to set the _basePath_ of the [RoutesConfig](https://tsoa-community.github.io/reference/interfaces/routesconfig.html)

**Comment by zenmaster15** [comment]
> E: You could try to pass a router instead of app to RegisterRoutes for science, but that would likely > mean your OAS would be incorrect.

Yeah, i tried passing a router to RegisterRoutes by using a custom handlebar file (just editing the type of the argument) and it works fine. Anyways I found out that i can use an express app itself as a middleware and ended up using that instead of routers. Support for using routers would be much appreciated.

I've ended up with the following folder structure
 -src
    -v1
       -controllerV1
       -index.ts
       -tsoa.json
    -v2
       -controllerV2
       -index.ts
       -tsoa.json
  -index.ts
I had to name the controller folders according to their version due to tsoa automatically finding any folder named controller in the directory tree and causes conflicts, even though i overridden it using the config
